### PR TITLE
feat(classifier): non-binding textlint suggestion for textlint-suitable rules (#215)

### LINKED ISSUE #215: Research: classifier external_check routing for textlint-suitable rules (#80 followup)
`docs/backend-external.md` states: "the classifier still has no \`external\` branch; rules use \`external_check\` only when authored that way (or compiled from a future adapter-aware classification pass)."

The classifier in `src/gate_keeper/classifier.py` currently has no logic to detect textlint-suitable patterns (grammar/spelling/terminology rules) and route them to `external_check`. Semantic rules about prose quality currently must be hand-authored as `external_check` rather than auto-classified.

## Research scope

This is an INVESTIGATION issue, not implementation. Deliverable: a comment on this issue with:

1. **Survey of existing rule patterns** that *should* route to external+textlint based on `docs/semantic-rules.md`'s mechanical-vs-semantic guidance (#208 docs landed here will help).
2. **Candidate heuristics**:
   - Rules matching "must not contain X" + concrete prose terms → external_check candidate
   - Rules matching "use the term Y instead of Z" → textlint terminology candidate
   - etc.
3. **Tradeoff analysis**: false positive risk of auto-routing vs explicit author opt-in.
4. **Recommendation**: ship classifier heuristics now (slice 1), defer to author opt-in, or hybrid?

## Out of scope

- Implementing the heuristics (separate impl slice if recommendation is "ship").
- Adding new RuleKind values.

## Refs

Refs: #164, #80, #95. Surfaced from Explore-3 architecture scan.

### PR DESCRIPTION
**Pull request by uda-lab-agent**
## Summary

Per the [owner decision on #215](https://github.com/t-uda/gate-keeper/issues/215), the classifier now detects textlint-suitable patterns and emits a **non-binding** ``suggested_backend`` annotation on the Rule IR. Effective routing (``backend_hint`` / ``kind``) is unchanged — the suggestion is advisory output only. Authors adopt a suggestion by setting ``backend_hint=external`` / ``kind=external_check`` in the rule IR explicitly; there is no auto-promotion.

### Heuristics (high precision, low recall)

- **Term-substitution / terminology enforcement** — "use X instead of Y", "must not use the term".
- **Named prose-style constraints** — passive voice, first-person pronouns, sentence length.
- **Capitalisation / spelling enforcement** — "must be capitalized as", "correct spelling".

### IR surface

The Rule IR gains three optional fields, all of which are omitted from ``to_dict()`` output when absent so existing fixtures stay byte-identical:

| Field | Type | Meaning |
| --- | --- | --- |
| ``suggested_backend`` | ``str \| None`` | e.g. ``"external+textlint"`` |
| ``suggestion_confidence`` | ``float \| None`` in ``[0.0, 1.0]`` | heuristic confidence |
| ``suggestion_rationale`` | ``str \| None`` | short human-readable explanation |

### Docs

``docs/backend-external.md`` gains a "Classifier suggestion (#215)" section that documents the advisory fields and the author opt-in workflow (edit IR JSON to set ``kind=external_check``, ``backend_hint=external``, ``params.tool=textlint``).

## Test plan

- [x] ``uv run pytest tests/test_classifier.py`` — 75 passed (16 new cases covering positive, negative, routing invariance, and IR round-trip).
- [x] ``uvx ruff check .`` / ``uvx ruff format --check .`` — clean.
- [x] ``uv run pyright`` — clean (project-config'd run).
- [x] Manual review: pre-existing LLM-rubric stub test failures on ``main`` (network-dependent, ``test_llm_rubric_backend.py::TestLlmRubricBackendStub``) are **unchanged** by this slice — they fail the same way before and after.

Closes #215

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by t-uda** [comment]
@codex please review this PR

**Comment by t-uda** [comment]
@codex please review this PR

File: src/gate_keeper/models.py [hunk 1]
```diff
@@ -184,13 +194,33 @@ def from_dict(cls, data: Any) -> Rule:
...
+            suggestion_confidence: float | None = None
+        else:
+            if isinstance(suggestion_confidence_raw, bool) or not isinstance(
+                suggestion_confidence_raw, (int, float)
+            ):
+                raise ValueError(
+                    f"Rule.suggestion_confidence: expected float, "
+                    f"got {type(suggestion_confidence_raw).__name__}"
+                )
+            suggestion_confidence = float(suggestion_confidence_raw)
```
[the remark this answers is not in the document]
Comment (uda-lab-agent) [reply src/gate_keeper/models.py:227]: Addressed in cb81582: Rule.from_dict now rejects non-finite values (NaN / Inf) and out-of-range numbers (outside [0.0, 1.0]) with a clear ValueError. Boundary values (0.0 and 1.0) remain accepted. 8 new test cases in TestTextlintSuggestion cover the boundary, out-of-range, non-finite, non-numeric, and boolean rejections.
